src/ngraph/runtime/backend_manager.cpp

```diff
@@ -111,21 +111,9 @@ shared_ptr<runtime::Backend> runtime::BackendManager::create_backend(const std::
             throw runtime_error("Backend '" + type + "' does not implement new_backend");
         }
 
-        function<void(runtime::Backend*)> delete_backend =
-            reinterpret_cast<void (*)(runtime::Backend*)>(DLSYM(handle, "delete_backend"));
-        if (!delete_backend)
-        {
-            CLOSE_LIBRARY(handle);
-            throw runtime_error("Backend '" + type + "' does not implement delete_backend");
-        }
-
-        runtime::Backend* backend = new_backend(config.c_str());
-        rc = shared_ptr<runtime::Backend>(backend, [=](runtime::Backend* b) {
-            delete_backend(b);
-            // CLOSE_LIBRARY(handle);
-        });
+        backend = new_backend(config.c_str());
     }
-    return rc;
+    return unique_ptr<runtime::Backend>(backend);
```

Comment:
One minor note -- if we ever want to support unloading a backend, we won't be able to do it via adding a deleter to the shared pointer.  But it's probably better to use a wrapper for that anyway, so we can defer unload as long as there are any outstanding objects.  (And it might be best to never unload backends; unloading dlls is tricky.)

Reply:
I was disappointed to find that unique_ptr does not handle deleters the same way that shared_ptr does and I could not find a way to add a deleter that I could pass without having to add another template arg to every reference of unique_ptr.

Reply:
There's always the hack of specializing `std::default_delete<>`, but I don't think it'd make sense for something like this.  :-)